File: src/main/app/src/app/service/customer-service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Customer } from '../model/Customer';

@Injectable({
  providedIn: 'root'
})
export class CustomerService {
  private baseUrl = 'http://localhost:8080/api/customers';

  constructor(private http: HttpClient) { }
  
  getCustomer(id: number): Observable<Customer> {
    return this.http.get<Customer>(`${this.baseUrl}/${id}`);
  }
 
  createCustomer(customer: Customer): Observable<Customer> {
    return this.http.post<Customer>(`${this.baseUrl}` + `/create`, customer);
  }
 
  updateCustomer(id: number, value: Customer): Observable<Customer> {
    return this.http.put<Customer>(`${this.baseUrl}/update/${id}`, value);
  }
 
  deleteCustomer(id: number): Observable<any> {
    return this.http.delete(`${this.baseUrl}/delete/${id}`, { responseType: 'text' });
  }
 
  getCustomersList(): Observable<any> {
    return this.http.get(`${this.baseUrl}/getAllCustomers`);
  }
 
  getCustomersByAge(age: number): Observable<any> {
    return this.http.get(`${this.baseUrl}/age/${age}`);
  }
  getCustomersById(id: number): Observable<any> {
    return this.http.get(`${this.baseUrl}/id/${id}`);
  }
  deleteAllCustomers(): Observable<any> {
    return this.http.delete(`${this.baseUrl}` + `/deleteAllCustomers`, { responseType: 'text' });
  }
  isActive(id: number, value: Customer): Observable<Customer> {
    return this.http.put<Customer>(`${this.baseUrl}/isActive/${id}`, value);
  }
  inActive(id: number, value: Customer): Observable<Customer> {
    return this.http.put<Customer>(`${this.baseUrl}/inActive/${id}`, value);
  }
}

File: src/main/java/com/curdcustomer/service/CustomerService.java
package com.curdcustomer.service;

import java.util.List;
import java.util.Optional;

import org.springframework.http.ResponseEntity;

import com.curdcustomer.model.Customer;

public interface CustomerService {
	Customer create(Customer customer);
	
	ResponseEntity<Customer> update(Customer customer, Long id);
	
	ResponseEntity<Customer> isActive(Customer customer, Long id);
	
	ResponseEntity<Customer> inActive(Customer customer, Long id);
	
	List<Customer> getAllCustomers();
	
	ResponseEntity<String> deleteAllCustomers();
	
	ResponseEntity<String>  deleteCustomer(long id);
	
	Optional<Customer> findById(long id);
	
	List<Customer> findByAge(int age);
}
